transformations/token_replacement/transformation.py

```diff
@@ -0,0 +1,129 @@
+import random
+import spacy
+
+from typing import List
+
+from initialize import spacy_nlp
+
+from interfaces.SentenceOperation import SentenceOperation
+
+from tasks.TaskTypes import TaskType
+
+from transformations.token_replacement.lookup_table_utils import (
+    load_lookup, 
+    get_token_replacement,
+)
+
+
+"""
+Implementation of the token replacement perturbation using lookup tables 
+with valid replacement candidates
+"""
+  
+class TokenReplacement(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(
+        self, seed=0, max_outputs = 1, replacement_prob = 0.2, min_length = 3, 
+        max_dist = 1, lookup = ["typos.xz", "ocr.xz"]):
+        """
+        :param seed: random seed
+        :param max_outputs: max. number of outputs to generate
+        :param replacement_prob: replacement probability [0.0, 1.0]
+        :param min_length: min. length of token that can be perturbed
+        :param max_dist: max. Levenshtein distance between the original and the
+                         perturbed token
+        :param lookup: dictionary or a list of files with lookup tables
+        """
+        super().__init__(seed, max_outputs=max_outputs)
+        #self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")
+        self.nlp = spacy.load("en_core_web_sm")
```

Comment:
Also one minor suggestion to reduce memory. This line can be replaced with the global spacy:
`from initialize import spacy_nlp`
`self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")`

Reply:
Thank you @kaustubhdhole for this suggestion!

If I use the global spacy object, the pytest test fails (the same problem as described here: #209). 

I've applied the fix suggested [here](https://github.com/GEM-benchmark/NL-Augmenter/issues/209#issuecomment-900407032) (thanks @AbinayaM02!) to reset the tokenizer of the global object in the generate method, so my transformation can work as expected. Still, I need to restore the old tokenizer afterwards, so other transformations can pass the pytest tests too. I suspect that some test.json files were generated using the already modified tokenizer and they fail when the original tokenizer is used. 

Please look into the transformation.py file for the workaround I've built in. I feel that this issue concerns many other transformation. I hope that @AbinayaM02 comes up with a nice solution for this issue :)